### PR TITLE
Removing Caddy and SSL + Making blueprints Traefik compatible

### PR DESCRIPTION
**Pull request by Ornias1993**
# Pull Request Template
### Purpose
The current version of Caddy is depricted. For that reason we needed either to move to Caddy 2.0 or to another system.

With Traefik getting into shape, It is going to replace caddy as the SSL solution. Meaning: It will be the ONLY ssl solution within Jailman. It will be a core part of jailman.

This means that we won't be using any of the more advanced features of Caddy.

Hence we will be removing two things:
- SSL from most jails (those where it is done by our scripts or easy to disable)
- Caddy

This will be replaced with:
- NGINX
- Easy to use Traefik configuration

### Notes:
Please be aware that our old solution doesn't even support letsencrypt or good certificates in a lot of cases. Even without fully setting up traefik, it would end-up giving jails the same self-signed certificates as we are generating now.

This also removes custom licencing for the Nextcloud Blueprint.
As all of the copyrightable code has by now be replaced by code from my hand and our licence is actually more permissive than the original gplv3 licence.

### Please make sure you have followed the self checks below before submitting a PR:

- [ ] Code is sufficiently commented
- [ ] Code is indented with tabs and not spaces
- [ ] The PR does not bring up any new errors
- [ ] The PR has been tested
- [ ] Any new files are named using lowercase (to avoid issues on case sensitive file systems)

### CODE REVIEW

**Comment by Ornias1993** [comment]
Fucking ducky... nextcloud actually kinda worked first test...
(at least PHP processing worked out fine)

**Comment by Ornias1993** [comment]
Custom traefik configs for blueprints that need one are added.
Remaining blueprints should work fine without custom config...

**Comment by Ornias1993** [comment]
Opening it up for review.
Traefik testing not (yet) done, but caddy removal is fully tested to be working

File: includes/global.sh [hunk 1]
```diff
@@ -222,7 +222,11 @@ if [ "${traefikstatus}" = "preinstalled" ]; then
 	# replace placeholder values.
 	sed -i '' "s|placeholderdashboardhost|${domain_name//&/\\&}|" /mnt/"${global_dataset_config}"/"${link_traefik}"/temp/"${1}".toml
 	sed -i '' "s|placeholdername|${1//&/\\&}|" /mnt/"${global_dataset_config}"/"${link_traefik}"/temp/"${1}".toml
-	sed -i '' "s|placeholderurl|${jail_ip}:${traefik_service_port}|" /mnt/"${global_dataset_config}"/"${link_traefik}"/temp/"${1}".toml
+	if [ -z "${traefik_service_port}" ]; then
+		sed -i '' "s|placeholderurl|${jail_ip}|" /mnt/"${global_dataset_config}"/"${link_traefik}"/temp/"${1}".toml
```
Comment (asmod3us) [inline includes/global.sh:226]: use traefik_temp for a shorter path?

Comment (Ornias1993) [reply includes/global.sh:226]: Did you check if it's even rebased yet? ;)

**Comment by Ornias1993** [comment]
My workstation died, should be able to continue next week... :(

**Comment by asmod3us** [comment]
> My workstation died, should be able to continue next week... :(

👎 How annoying! Hope you can resurrect it

**Comment by Ornias1993** [comment]
> > My workstation died, should be able to continue next week... :(
> 
> 👎 How annoying! Hope you can resurrect it

980ti died, I think its 5+ years old by now... has had 3 different coolers by now... has been watercooled, overclocked, aircooled overclocked again...

I think it served it's life well :)

But damn annoying, Hope the 1080ti upgrade makes me forget the pain of working on laptop for a week or so ;)

**Comment by Ornias1993** [comment]
Note: #182 needs to be dealth with when I get back to this!

**Comment by Ornias1993** [comment]
After 1.2.3 is released, I first need to re-check dev to make sure it still works (after all the rebases) and then i'll rebase this and finish it.

**Comment by Ornias1993** [comment]
I finally managed to get this rebased. 
Taking into account the Dev branch needs heavy retesting and bugfixing anyhow, i'm going to merge this into Dev, fix all the bugs at once and return to the traefik project in a third PR.

**Pull request merged**